test/functorch/test_eager_transforms.py

```diff
@@ -3204,13 +3204,9 @@ def f(x):
 
         B = 5
         x = torch.randn(B, 3)
-        with self.assertRaises(RuntimeError):
+        with self.assertRaisesRegex(RuntimeError, "Batching rule not implemented for aten::_make_dual"):
             vmap(f)(x)
 
-        x = torch.randn([])
-        with self.assertRaises(RuntimeError):
-            grad(f)(x)
```

Comment:
The test used to fail due to this issue : 

```python
import torch
from torch.func import jacfwd, jacrev, vmap, vjp, jvp, grad
from functorch import make_fx
from torch._C._functorch import unwrap_if_dead, is_dead_tensor_wrapper
from torch.autograd.forward_ad import make_dual, dual_level

torch.manual_seed(420)

x = torch.randn(())

def f(x):
    y = torch.autograd.functional.jacobian(
                lambda x: x.sin().sum(), x, strategy='forward-mode', vectorize=True)
    return y

grad(f)(x)

```

Output:
```
RuntimeError: unwrapped_count > 0 INTERNAL ASSERT FAILED at "/home/kshiteej/Pytorch/pytorch_functorch/aten/src/ATen/functorch/TensorWrapper.cpp":213, please report a bug to PyTorch. Should have at least one dead wrapper
```

Reply:
Also, do you know why this succeeds now? We are not creating a new torch.tensor using a list of tensors, so it sounds like there is some interesting interaction going on

Reply:
It looks to be happening because of a bad interaction with legacy `_vmap`. If I update it to `torch.func.vmap`, the code fails with `Batching rule not implemented for make_dual`.

https://github.com/pytorch/pytorch/blob/c377a8590b9b654a2cf03c005f5171ca63c38534/torch/autograd/functional.py#L476-L477

Also, this doesn't really succeed in a workable fashion. In the following script, it returns BatchedTensor which fails if you try to print it or compare it.

```python
import torch
from torch.func import jacfwd, jacrev, vmap, vjp, jvp, grad
from functorch import make_fx
from torch._C._functorch import unwrap_if_dead, is_dead_tensor_wrapper
from torch.autograd.forward_ad import make_dual, dual_level

torch.manual_seed(420)

x = torch.randn(())

def f(x):
    y = torch.autograd.functional.jacobian(
                lambda x: x.sin().sum(), x, strategy='forward-mode', vectorize=True)
    return y

def f_exp(x):
    y = jacrev(lambda x: x.sin().sum())(x)
    return y

j = jacrev(f)(x)
# print(j)  # RuntimeError: Batching rule not implemented for aten::is_nonzero. We could not generate a fallback.

expected_j = jacrev(f_exp)(x)
print(expected_j)  # Works

torch.testing.assert_close(j, expected_j)  # RuntimeError: Batching rule not implemented for aten::is_nonzero. We could not generate a fallback.
```